### DISCUSSION TITLE
Gmail refuses mails send by PHPMailer

### DISCUSSION THREAD

**Discussion by rageusen** [PHPMailer/PHPMailer#2832] [Q&A]
Recently Gmail started refusing mails send by PHPMailer because of duplicate headers. Other mail clients still accept the mails without an issue. What am I doing wrong?

This is the Gmail message:
----- The following addresses had permanent fatal errors ----- <[xxx@gmail.com](mailto:xxx@gmail.com)>
    (reason: 550-5.7.1 [157.193.71.2] This message is not RFC 5322 compliant, the issue is: 550-5.7.1 duplicate T...2 specifications for more information. di7-20020a170906730700b007b2a3ee2d2asi13744069ejc.703 - gsmtp)

   ----- Transcript of session follows ----- ... while talking to gmail-smtp-in.l.google.com.:
>>> DATA
<<< 550-5.7.1 [157.193.71.2] This message is not RFC 5322 compliant, the issue is:
<<< 550-5.7.1 duplicate To headers. To reduce the amount of spam sent to Gmail, <<< 550-5.7.1 this message has been blocked. Please review <<< 550 5.7.1  RFC 5322 specifications for more information. di7-20020a170906730700b007b2a3ee2d2asi13744069ejc.703 - gsmtp
550 5.7.1 [157.193.71.2] This message is not RFC 5322 compliant, the issue is: 550-5.7.1 duplicate To headers. To reduce the amount of spam sent to Gmail, 550-5.7.1 this message has been blocked. Please review 550 5.7.1  RFC 5322 specifications for more information. di7-20020a170906730700b007b2a3ee2d2asi13744069ejc.703 - gsmtp

And this is my code for sending the mails:
```php
    $objMail = new PHPMailer();
    $objMail->IsHTML(true);
    $objMail->From = cAPI::global('email:afzender');
    $objMail->FromName = cAPI::global('email:afzender:naam');
    $objMail->CharSet = 'UTF-8';
    $objMail->Subject = str_replace(array_keys($arrSR), array_values($arrSR), $strSubject );
    $objMail->Body = str_replace(array_keys($arrSR), array_values($arrSR), $strBody );

    if ( cAPI::config('debug') ) {
        $objMail->addAddress( cAPI::config('debug_mail') );
    } else {
        $objBegeleiders = PDO_DataObject::factory("tbl_studadmin_thesis_begeleiders");
        $objBegeleiders->begeleider_thesis_id = $_id;
        $objBegeleiders->begeleider_functie = "promotor";
        $objBegeleiders->find();
        while ($objBegeleiders->fetch()) {
            $objMail->addBcc($objBegeleiders->begeleider_email);
        }
    }

    if (!$objMail->send())
        return false;
    else
        return true;
```

**Comment by Synchro** [comment]
This is an intermittent problem with the PHP `mail()` function that is used behind the scenes in your code. You can work around it by using SMTP to localhost instead, which is also safer and faster. Add this line to your code:

    $mail->isSMTP();

**Reply by rageusen** [reply]
Solved the problem thanks to your tip!

**Comment by Richard2012** [comment]
I have the same problem.
Using the example script I can successfully  send mail to many different domains with the **exception of gmail.com**
Have tried both 587 and 465 ports.
Have $mail->isSMTP()
I am using Hostgator mail server.
The worst part is that I am getting success message (but nothing in gmail inbox nor in spambox).

**Reply by Wisdom19Wisdom** [reply]
I'm having the same using.., but I'm using XAMPP

**Comment by decomplexity** [comment]
A long shot, but do you have an SPF  TXT record set in your domain's DNS? Google recently tightened up inbound mail security and may be silently blackholing (instead of rejecting as spam) mail from non-SPF sources.

**Reply by Richard2012** [reply]
Thank you, your suggestion helped to solve the problem. I works now. Hostgator has a cPanel interface to manage the SPF TXT.
Perhaps the hint cold be included in the phpmailer documentation?

**Reply by Synchro** [reply]
Feel free to add a note - the wiki is open to edits.
